### PR TITLE
feat: [PL-22399]: Made a component for SCM validation and info

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
##### Summary:
Before we had only one central place for all the git commits for all entities so SCM check was added inline in [SaveToGitDialog](https://github.com/harness/harness-core-ui/pull/3106) . Now, with full sync we need to add SCM check in 2 more modals so making a separate component became a logical requirement.

##### Jira Links:
https://harness.atlassian.net/browse/PL-22399

##### Screenshots:
New code sample integration experience : 

**With SCM** 
Screenshot 2022-02-07 at 7 05 40 PM

**Without SCM**
Screenshot 2022-02-07 at 7 07 58 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by avinashmadhwani02 — DISMISSED** [review]

File: src/modules/10-common/components/SCMCheck/SCMCheck.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import paths from '@common/RouteDefinitions'
+import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
+import { useStrings } from 'framework/strings/String'
+import css from './SCMCheck.module.scss'
+
+export interface SCMCheckProps {
+  profileLinkClickHandler?: () => void // If used inside modal, this can be use to close modal
+  title?: string
+  className?: string
+  validateSCM: React.Dispatch<React.SetStateAction<boolean>> // Can be used to disable the commit
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/SCMCheck/SCMCheck.tsx:24]: is this is a setstate function? like we discussed earlier, setState should not be passed to other components

File: src/modules/10-common/components/SCMCheck/SCMCheck.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    setCurrentUserInfo(user)
+    validateSCM(!loadingCodeManagers && user)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [codeManagers])
+
+  return (
+    <Container className={cx(className)} margin={{ bottom: 'large' }}>
+      {currentUserInfo ? (
+        <Layout.Horizontal className={css.userInfo} flex={{ alignItems: 'center' }}>
+          <Avatar size="small" name={currentLoggedInUser.name} backgroundColor={Color.PRIMARY_7} hoverCard={false} />
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/SCMCheck/SCMCheck.tsx:48]: name might be empty. fallback to email in that case

File: src/modules/10-common/components/SCMCheck/SCMCheck.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          <Text color={Color.GREY_700}>
+            {getString('common.git.currentUserLabel', { user: currentLoggedInUser.name })}
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/components/SCMCheck/SCMCheck.tsx:50]: same here

File: src/modules/10-common/components/SCMCheck/SCMCheck.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/10-common/components/SCMCheck/SCMCheck.tsx]: corrected.

File: src/modules/10-common/components/SCMCheck/SCMCheck.tsx (hunk 2 above)
Comment (sunnykesh-harness) [reply src/modules/10-common/components/SCMCheck/SCMCheck.tsx]: done

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**